src/irmin-git/contents.ml

```diff
@@ -52,9 +52,10 @@ module Make (G : Git.S) (C : Irmin.Contents.S) = struct
       match Raw.of_raw_with_header ~off buf with
       | Ok g -> (
           match V.of_git g with
-          | Some g ->
-              pos_ref := String.length buf;
-              g
+          | Some v ->
+              let len = Raw.length_with_header g in
+              pos_ref := Int64.of_int off |> Int64.add len |> Int64.to_int;
```

Comment:
Perhaps it's simpler to cast `len` to an int? (as `Int64.to_int` doesn't provide any guarantee that we didn't overflow anyway)